server/service/src/invoice/inbound_shipment/update/generate.rs

```diff
@@ -126,23 +120,33 @@ pub(crate) fn generate(
         None
     };
 
+    let update_donor = if should_update_donor(input_donor_id, &patch.update_donor_method)
```

Comment:
Moved update here since we want to update donor every time it's changed... wondering if we should add to the `generate_line_and_stock_line` function... Our frontend doesn't allow change without prompting reason, but if someone else uses our API... 👀